src/Lykke.Service.MarketMakerReports.Core/Domain/InventorySnapshots/AssetBalanceInventory.cs

```diff
@@ -1,13 +1,34 @@
+using System;
 using System.Collections.Generic;
+using System.Linq;
+using Newtonsoft.Json;
 
 namespace Lykke.Service.MarketMakerReports.Core.Domain.InventorySnapshots
 {
     public class AssetBalanceInventory
     {
-        public string Asset { get; set; }
+        [JsonProperty("Asset")]
+        public string AssetId { get; set; }
+        
+        public string AssetDisplayId { get; set; }
         
         public IReadOnlyList<AssetInventory> Inventories { get; set; }
         
         public IReadOnlyList<AssetBalance> Balances { get; set; }
+
+        public AssetBalance GetBalanceByExchange(string exchange)
+        {
+            return Balances.SingleOrDefault(x => string.Equals(x.Exchange, exchange, StringComparison.InvariantCultureIgnoreCase)) 
+                   ?? new AssetBalance
+                        {
+                            Exchange = exchange
+                        };
+        }
+
+        public decimal TotalBalance => Balances.Sum(x => x.Amount);
```

Comment:
The "ArgumentNullException" possible if balances is empty. Add "DefaultIfEmpty(0)" to avoid exception.

Reply:
There will not be exception in case the list is empty, but for case the list is null I've added a special cases into the mapping.